packages/notte-agent/src/notte_agent/falco/agent.py

```diff
@@ -305,3 +345,18 @@ async def _run(self, task: str, url: str | None = None) -> AgentResponse:
         logger.info(f"🚨 {error_msg}")
         notte_core.set_error_mode("developer")
         return self.output(error_msg, False)
+    
+    
+    
+    def _get_valid_action_set(self, last_obs: Observation) -> set[str]:
```

Comment:
i think it's probably best to add a function in the `Observation` class that's already there (e.g. `obs.is_action_valid(...)`